description/src/transformer.rs

```diff
@@ -70,3 +75,46 @@ where
         Ok(r)
     }
 }
+
+///
+/// ## Background:
+///
+/// There is a problem in generating recursive type descriptions:
+/// Suppose we have the following setup:
+/// ```rust
+/// struct A {
+///     bees: Vec<B>
+/// }
+///
+/// struct B {
+///     id: u8,
+///     others: Vec<B>
+/// }
+/// ```
+/// This could be described as:
+/// ```txt,no_run
+/// struct A {
+///     bees: Vec<struct B {
+///         id: u8,
+///         others: Vec<B>
+///     }>
+/// }
+/// ```
+/// But the recursive resolving would get stuck in the middle, reporting recursion.
```

Comment:
Would something like this solve the issue?

- keep a cache of visited types
- when we encounter a recursion in describing a type, we simply resolve the type to the type's name, iff they match?

```rust
struct B { -> We are describing type B
id: u8,
other: Vec<B> -> Encountered type B name, resolve to the name, because this B type name == B the type name we are currently describing
}
```